metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -0,0 +1,679 @@
+from sqlalchemy.engine.reflection import Inspector
+
+from datahub.configuration.common import AllowDenyPattern
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import (
+    SourceCapability,
+    SupportStatus,
+    capability,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.sql.sql_common import (
+    SQLAlchemySource,
+    SqlWorkUnit,
+    register_custom_type,
+)
+from datahub.ingestion.source.sql.sql_config import (
+    BasicSQLAlchemyConfig,
+    make_sqlalchemy_uri,
+    SQLAlchemyConfig,
+)
+from datahub.metadata.schema_classes import (
+    BooleanTypeClass,
+    ChangeTypeClass,
+    UnionTypeClass,
+)
+from sqlalchemy.exc import ProgrammingError, ResourceClosedError
+
+from .job_models import (
+    JobStep,
+    MSSQLDataFlow,
+    MSSQLDataJob,
+    MSSQLJob,
+    MSSQLProceduresContainer,
+    ProcedureDependency,
+    ProcedureLineageStream,
+    ProcedureParameter,
+    StoredProcedure,
+)
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+register_custom_type(sqlalchemy.dialects.mssql.BIT, BooleanTypeClass)
+register_custom_type(sqlalchemy.dialects.mssql.SQL_VARIANT, UnionTypeClass)
+
+
+class SQLServerConfig(BasicSQLAlchemyConfig):
+    # defaults
+    host_port: str = Field(default="localhost:1433", description="MSSQL host URL.")
+    scheme: str = Field(default="mssql+pytds", description="", hidden_from_schema=True)
+    include_stored_procedures: bool = Field(
+        default=True, description="Include ingest of stored procedures. Requires access to the 'sys' schema."
+    )
+    include_stored_procedures_code: bool = Field(
+        default=True, description="Include information about object code."
+    )
+    include_jobs: bool = Field(
+        default=True, description="Include ingest of MSSQL Jobs. Requires access to the 'msdb' and 'sys' schema."
+    )
+    include_descriptions: bool = Field(
+        default=True, description="Include table descriptions information."
+    )
+    use_odbc: bool = Field(
+        default=False,
+        description="See https://docs.sqlalchemy.org/en/14/dialects/mssql.html#module-sqlalchemy.dialects.mssql.pyodbc.",
+    )
+    uri_args: Dict[str, str] = Field(
+        default={},
+        description="Arguments to URL-encode when connecting. See https://docs.microsoft.com/en-us/sql/connect/odbc/dsn-connection-string-attribute?view=sql-server-ver15.",
+    )
+    database_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for databases to filter in ingestion.",
+    )
+    database: Optional[str] = Field(
+        default=None,
+        description="database (catalog). If set to Null, all databases will be considered for ingestion.",
+    )
+
+    database_alias: Optional[str] = Field(
```

Comment:
this should be removed from here. Its refractored and moved to the superclass `BasicSQLAlchemyConfig` of this config ([here](https://github.com/datahub-project/datahub/blob/0cdb5e4b4b77c741660c88266ef478c4e8a3a4cc/metadata-ingestion/src/datahub/ingestion/source/sql/sql_config.py#L93)).

Reply:
Removed